scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -125,127 +144,58 @@ fun Bitmap.fragment(
-        ),
-        Rect(
-            0,
-            (this.height / 2) + heightOffset,
-            (this.width / 2) - widthOffset,
-            this.height
-        ),
-        Rect(
-            (this.width / 2) - widthOffset,
-            (this.height / 2) + heightOffset,
-            (this.width / 2) + widthOffset,
-            this.height
-        ),
-        Rect(
-            (this.width / 2) + widthOffset,
-            (this.height / 2) + heightOffset,
-            this.width,
-            this.height
-        )
-    )
-    val toSegments = arrayOf(
-        Rect(
-            0,
-            0,
-            toBorderWidth,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth,
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            0,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth
-        ),
-        Rect(
-            0,
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            112,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension),
-            (2 * toBorderWidth + toCenterDimension)
-        )
+    // Transforms the viewfinder from the preview to the image size
+    val projectedViewFinder = previewSize.projectRegionOfInterest(toSize = this.size(), regionOfInterest = cardFinder)
+    // Creates a square version of the viewfinder
+    val projectViewFinderGreatestDimension = max(projectedViewFinder.width(), projectedViewFinder.height())
+    val projectedViewFinderSquare = Size(projectViewFinderGreatestDimension, projectViewFinderGreatestDimension)
+
+    require(this.size().toRect().contains(projectedViewFinderSquare.centerOn(projectedViewFinder))) {
+        "Viewfinder too close to edge of the screen"
+    }
+
+    val aspectRatio = 9.0f / 16.0f
+    // Creates a rect of a certain aspect ratio surrounding the view finder
+    val aspectRatioCrop = minAspectRatioSurroundingSize(projectedViewFinderSquare, aspectRatio).centerOn(projectedViewFinder)
```

Comment:
this should probably happen outside of zoom. it's a separate step that should occur before we zoom in.